diff --git a/src/ptex/PtexInt.h b/src/ptex/PtexInt.h
index cae8743..c03e5dd 100644
--- a/src/ptex/PtexInt.h
+++ b/src/ptex/PtexInt.h
@@ -41,7 +41,9 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
   @brief Portable fixed-width integer types
 */
 
-#if defined(_WIN32) || defined(_WINDOWS) || defined(_MSC_VER)
+#if defined(__MINGW32__) || defined(__MINGW64__)
+#include <stdint.h>
+#elif defined(_WIN32) || defined(_WINDOWS) || defined(_MSC_VER)
 
 #if defined(_MSC_VER) && _MSC_VER >= 1600
 #include <stdint.h>
